group/assignment6/RearrangeCars.cpp

```diff
@@ -0,0 +1,221 @@
+#include "RearrangeCars.h"
+
+RearrangeCars::RearrangeCars(const vector<int> &startStates) : startStates(startStates) {}
+
+void RearrangeCars::setStartStates(const vector<int> &startStates) {
+      this->startStates = startStates;
+}
+
+int RearrangeCars::findEmptySlotId() {
+      for (int slotId = 0; slotId < this->startStates.size(); slotId++) {
+            if (this->startStates[slotId] == EMPTY_SLOT_ID)
+                  return slotId;                // no car has EMPTY_SLOT_ID id, it is reserved for empty slot

```

Comment:
I find our that lines that are non-obvious should get a comment at the end of the line. These end-of-line comments should be separated from the code by 2 spaces.